File: 22 - Find the Smallest Element in Rotated Sorted Array/solution.js
function findSmallest(nums) {
  // Your code here
  let start = 0;
  let end = nums.length - 1;

  while (start < end) {
    const mid = Math.floor((start + end) / 2);

    if (nums[mid] > nums[end]) {
      start = mid + 1;
    } else {
      end = mid;
    }
  }

  return nums[start];
}

// Test cases
console.log(findSmallest([4, 5, 6, 7, 0, 1, 2]));

File: 22 - Find the Smallest Element in Rotated Sorted Array/solution.py
def find_smallest(nums):
    # Your code here
    start = 0
    end = len(nums) - 1

    while start < end:
        mid = (start + end) // 2

        if nums[mid] > nums[end]:
            start = mid + 1
        else:
            end = mid

    return nums[start]


#   Test cases
print(find_smallest([4, 5, 6, 7, 0, 1, 2]))